slick/src/main/scala/slick/jdbc/JdbcDataSource.scala

```diff
@@ -173,6 +178,24 @@ object HikariCPJdbcDataSource extends JdbcDataSourceFactory {
     val ds = new HikariDataSource(hconf)
     new HikariCPJdbcDataSource(ds, hconf)
   }
+
+  def extractUrl(maybeUrl: Option[String]): (Option[String], Option[(String, String)]) = {
+    maybeUrl match {
+      case Some(PostgresFullUrl(username, password, host, dbname)) =>
+        Some(s"jdbc:postgresql://$host/$dbname") -> Some(username -> password)
+
+      case Some(url @ MysqlFullUrl(username, password, host, dbname)) =>
+        val defaultProperties = "?useUnicode=yes&characterEncoding=UTF-8&connectionCollation=utf8_general_ci"
```

Comment:
This bug is basically a regression. On the mailing list, I pointed out that it's not only broken for Heroku/Postgres, but that Play did some more magic (e.g. setting the stuff for MySQL) that has been lost. You may ask the Play devs why this is there or just remove it and wait if someone complains. Here is what I wrote on the mailing list, including some digging into the Play git history:

> Note that the Play Connection Pool did some more stuff for MySQL and H2 [0]. I'm not sure if that is still relevant - the H2 stuff was last edited in 2013 [1], the MySQL stuff in 2012 [2]. The main problem I'm seeing here is - if that code is still needed for current versions of MySQL and H2 - that if someone tries to migrate from anorm to Slick, or from Play 2.3 to 2.4, their previously fine shortcut URL will now fail for strange reasons.
> 
> [0] https://github.com/playframework/playframework/blob/5d7adcbc246323310f6c9a9ef26555902e0ba3dc/framework/src/play-jdbc/src/main/scala/play/api/db/ConnectionPool.scala#L66-L72
> [1] https://github.com/playframework/playframework/commit/26e7dca3c0aaae74ec18bcc8e03c8dc90d87f8f4
> [2] https://github.com/playframework/playframework/commit/0906721b4cdf1ed6b7095d6dbd8afa658ec28bc2